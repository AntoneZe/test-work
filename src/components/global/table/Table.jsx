import React from "react";

import Button from "../button/Button";

import "./table.less";

const Table = () => {
  return (
    <div className="custom-table__wrapper">
      <table className="custom-table">
        <thead>
          <tr>
            <th>id</th>
            <th>Имя</th>
            <th>Номер телефона</th>
            <th></th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>1</td>
            <td>Leanne Graham</td>
            <td>1-770-736-8031</td>
            <td>
              <Button onClick={() => {}}>
                <img src="public/img/delete.svg" alt="delete" />
              </Button>
            </td>
          </tr>
          <tr>
            <td>2</td>
            <td>Ervin Howell</td>
            <td>010-692-6593</td>
            <td>
              <Button onClick={() => {}}>
                <img src="public/img/delete.svg" alt="delete" />
              </Button>
            </td>
          </tr>
          <tr>
            <td>3</td>
            <td>Clementine Bauch</td>
            <td>1-463-123-4447</td>
            <td>
              <Button onClick={() => {}}>
                <img src="public/img/delete.svg" alt="delete" />
              </Button>
            </td>
          </tr>
          <tr>
            <td>4</td>
            <td>Patricia Lebsack</td>
            <td>493-170-9623</td>
            <td>
              <Button onClick={() => {}}>
                <img src="public/img/delete.svg" alt="delete" />
              </Button>
            </td>
          </tr>
          <tr>
            <td>5</td>
            <td>Chelsey Dietrich</td>
            <td>(254)954-1289</td>
            <td>
              <Button onClick={() => {}}>
                <img src="public/img/delete.svg" alt="delete" />
              </Button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  );
};

export default Table;
