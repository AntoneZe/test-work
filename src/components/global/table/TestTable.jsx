import React from "react";

import Table from "./Table";

const titleList = ["id", "name", "phone"];
const rows = [
  { id: 1, name: "test-1", phone: "11" },
  { id: 2, name: "test-2", phone: "22" },
  { id: 3, name: "test-3", phone: "33" },
];
const deleteCallback = () => {
  return "done";
};

const TestTable = () => {
  return (
    <Table titleList={titleList} rows={rows} deleteCallback={deleteCallback} />
  );
};

export default TestTable;
