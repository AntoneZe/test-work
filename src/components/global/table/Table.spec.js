import React from "react";

import TestTable from "./TestTable";
import Table from "./Table";

describe("Table component", () => {
  it("should render Table component", () => {
    const component = render(<TestTable />);
    expect(component).toMatchSnapshot();
  });
});

describe("Table component with deleteCallback", () => {
  it("should call callback once", () => {
    const mockCallBack = jest.fn();

    const event = {
      preventDefault() {},
      target: {
        id: "test-id",
      },
    };

    const component = mount(<TestTable deleteCallback={mockCallBack} />);
    const btn = component.find("button").first();
    btn.simulate("click", event);
    expect(mockCallBack.mock.calls.length).toEqual(0);
  });
});

describe("Table with empty row list", () => {
  it("should render with tbody 'no-data' message", () => {
    const component = mount(
      <Table rows={[]} titleList={["test-1", "test-2", "test-3"]} />
    );
    expect(component).toMatchSnapshot();
  });
});
