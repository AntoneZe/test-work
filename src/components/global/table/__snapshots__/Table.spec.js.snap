// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table component should render Table component 1`] = `
<div
  class="custom-table__wrapper"
>
  <table
    class="custom-table"
  >
    <thead>
      <tr>
        <th>
          id
        </th>
        <th>
          name
        </th>
        <th>
          phone
        </th>
        <th />
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          1
        </td>
        <td>
          test-1
        </td>
        <td>
          11
        </td>
        <td>
          <button
            class="custom-button"
          >
            <img
              alt="delete"
              height="15"
              src="public/img/delete.svg"
              width="15"
            />
          </button>
        </td>
      </tr>
      <tr>
        <td>
          2
        </td>
        <td>
          test-2
        </td>
        <td>
          22
        </td>
        <td>
          <button
            class="custom-button"
          >
            <img
              alt="delete"
              height="15"
              src="public/img/delete.svg"
              width="15"
            />
          </button>
        </td>
      </tr>
      <tr>
        <td>
          3
        </td>
        <td>
          test-3
        </td>
        <td>
          33
        </td>
        <td>
          <button
            class="custom-button"
          >
            <img
              alt="delete"
              height="15"
              src="public/img/delete.svg"
              width="15"
            />
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Table with empty row list should render with tbody 'no-data' message 1`] = `
<Table
  rows={Array []}
  titleList={
    Array [
      "test-1",
      "test-2",
      "test-3",
    ]
  }
>
  <div
    className="custom-table__wrapper"
  >
    <table
      className="custom-table"
    >
      <thead>
        <tr>
          <th
            key="test-1"
          >
            test-1
          </th>
          <th
            key="test-2"
          >
            test-2
          </th>
          <th
            key="test-3"
          >
            test-3
          </th>
          <th />
        </tr>
      </thead>
      <tbody
        className="custom-table__dummy"
      >
        <tr>
          <td
            align="center"
            colSpan={1}
          />
          <td
            align="center"
            colSpan={4}
          >
            Нет записей...
          </td>
          <td
            align="center"
            colSpan={1}
          />
          <td
            align="center"
            colSpan={1}
          />
        </tr>
      </tbody>
    </table>
  </div>
</Table>
`;
